glean-core/ios/Glean/Glean.swift

```diff
@@ -263,6 +263,64 @@ public class Glean {
         }
     }
 
+    /// When applications are launched using the custom URL scheme, this helper function will process
+    /// the URL and parse the debug commands
+    ///
+    /// - parameters:
+    ///     * url: A `URL` object containing the Glean debug commands as query parameters
+    public func handleCustomUrl(url: URL) {
+        guard url.host?.removingPercentEncoding == "glean",
+            let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
+            let params = components.queryItems else {
+            logger.error("Error parsing query parameters")
+            return
+        }
+
+        var tagName: String?
+        var logPings: Bool?
+        var sendPing: String?
+
+        for param in params {
+            switch param.name {
+            case "tagPings":
+                if let pingTag = param.value,
+                    let pingTagPattern = try? NSRegularExpression(pattern: "^[a-zA-Z0-9-]{1,20}$") {
+                    _ = pingTagPattern.matches(
+                        in: pingTag,
+                        options: [],
+                        range: NSRange(location: 0, length: pingTag.count)
+                    ).map { _ in
+                        tagName = pingTag
+                        logger.debug("Pings tagged with: \(pingTag)")
+                    }
+
+                } else {
+                    logger.error("No value passed to tagPings")
+                }
+            case "logPings":
+                logPings = param.value == "true" ? true : false
+                logger.debug("Log pings set to: \(logPings!)")
+            case "sendPing":
+                sendPing = param.value
+            default:
+                logger.error("Unknown parameter passed to handleCustomUrl")
+            }
+        }
+
+        configuration = Configuration(
+            serverEndpoint: configuration?.serverEndpoint ?? Configuration.Constants.defaultTelemetryEndpoint,
+            userAgent: configuration?.userAgent ?? Configuration.Constants.defaultUserAgent,
+            logPings: logPings ?? configuration?.logPings ?? Configuration.Constants.defaultLogPings,
+            maxEvents: configuration?.maxEvents,
+            pingTag: tagName ?? configuration?.pingTag,
+            channel: configuration?.channel
+        )
```

Comment:
Do we really need to re-set the whole configuration object again and again? I find the whole inline `??` in this block unreadable.

Reply:
Well, because the configuration is filled with `let` constants, yes, I need to rebuild the entire configuration object.  If we relaxed some of the constant values to variables, or even if we made a copy constructor, it would simplify this.  Let me explore what I can do to simplify this...